[Bug]:   Application run failed - Error creating bean with name 'playQueueWSController'

### What happened?

Container stopping with Error after started , using docker in Unraid

![image](https://github.com/kagemomiji/airsonic-advanced/assets/28630321/6e05ee11-abdc-4e4c-a2a1-f7ba015eb787)
.
.
.
![image](https://github.com/kagemomiji/airsonic-advanced/assets/28630321/1f0b2afb-b404-4e37-a63c-643b2c87c85d)


### Steps to reproduce

Run Container lasted version

### Version

11.1.x (Edge)

### Version Detail

Edge Release 11.1.3-SNAPSHOT.20231009014635

### Operating System

Docker in Unraid (linux x64)

### Java Version

17.0.5+8

### Database

MariaDB 8.x

### DB Detail

MariaDB

### Configuration paramter

...

### Proxy Server

_No response_

### client detail

_No response_

### language

None

### Relevant log output

```console
20.10.2023
9:57:47
2023-10-20 09:57:47.976  INFO --- liquibase.util.ShowSummaryUtil           : Update summary generated
20.10.2023
9:57:47
2023-10-20 09:57:47.980  INFO --- l.l.StandardLockService                  : Successfully released change log lock
20.10.2023
9:57:47
2023-10-20 09:57:47.986  INFO --- l.l.StandardLockService                  : Successfully released change log lock
20.10.2023
9:57:47
2023-10-20 09:57:47.986  INFO --- liquibase.command.CommandScope           : Command execution complete
20.10.2023
9:57:50
2023-10-20 09:57:50.958  INFO --- o.a.p.s.TaskSchedulingService            : Task remote-playstatus-cleanup scheduled
20.10.2023
9:57:51
2023-10-20 09:57:51.032  INFO --- o.a.player.service.PlayerService         : Deleting old players
20.10.2023
9:57:51
2023-10-20 09:57:51.108  WARN --- ConfigServletWebServerApplicationContext : Exception encountered during context initialization - cancelling refresh attempt: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'playQueueWSController': Unsatisfied dependency expressed through field 'playerService'; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'playerService': Invocation of init method failed; nested exception is org.springframework.dao.InvalidDataAccessApiUsageException: No EntityManager with actual transaction available for current thread - cannot reliably process 'remove' call; nested exception is javax.persistence.TransactionRequiredException: No EntityManager with actual transaction available for current thread - cannot reliably process 'remove' call
20.10.2023
9:57:51
2023-10-20 09:57:51.276 ERROR --- o.s.boot.SpringApplication               : Application run failed
20.10.2023
9:57:51
20.10.2023
9:57:51
org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'playQueueWSController': Unsatisfied dependency expressed through field 'playerService'; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'playerService': Invocation of init method failed; nested exception is org.springframework.dao.InvalidDataAccessApiUsageException: No EntityManager with actual transaction available for current thread - cannot reliably process 'remove' call; nested exception is javax.persistence.TransactionRequiredException: No EntityManager with actual transaction available for current thread - cannot reliably process 'remove' call
20.10.2023
9:57:51
	at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor$AutowiredFieldElement.resolveFieldValue(AutowiredAnnotationBeanPostProcessor.java:713) ~[spring-beans-5.3.30.jar:5.3.30]
20.10.2023
9:57:51
	at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor$AutowiredFieldElement.inject(AutowiredAnnotationBeanPostProcessor.java:693) ~[spring-beans-5.3.30.jar:5.3.30]
20.10.2023
9:57:51
	at org.springframework.beans.factory.annotation.InjectionMetadata.inject(InjectionMetadata.java:119) ~[spring-beans-5.3.30.jar:5.3.30]
20.10.2023
9:57:51
	at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor.postProcessProperties(AutowiredAnnotationBeanPostProcessor.java:408) ~[spring-beans-5.3.30.jar:5.3.30]
20.10.2023
9:57:51
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.populateBean(AbstractAutowireCapableBeanFactory.java:1431) ~[spring-beans-5.3.30.jar:5.3.30]
20.10.2023
9:57:51
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:619) ~[spring-beans-5.3.30.jar:5.3.30]
20.10.2023
9:57:51
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:542) ~[spring-beans-5.3.30.jar:5.3.30]
20.10.2023
9:57:51
	at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:335) ~[spring-beans-5.3.30.jar:5.3.30]
20.10.2023
9:57:51
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:234) ~[spring-beans-5.3.30.jar:5.3.30]
20.10.2023
9:57:51
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:333) ~[spring-beans-5.3.30.jar:5.3.30]
20.10.2023
9:57:51
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:208) ~[spring-beans-5.3.30.jar:5.3.30]
20.10.2023
9:57:51
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingletons(DefaultListableBeanFactory.java:955) ~[spring-beans-5.3.30.jar:5.3.30]
20.10.2023
9:57:51
	at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:921) ~[spring-context-5.3.30.jar:5.3.30]
20.10.2023
9:57:51
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:583) ~[spring-context-5.3.30.jar:5.3.30]
20.10.2023
9:57:51
	at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.refresh(ServletWebServerApplicationContext.java:147) ~[spring-boot-2.7.16.jar:2.7.16]
20.10.2023
9:57:51
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:731) ~[spring-boot-2.7.16.jar:2.7.16]
20.10.2023
9:57:51
	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:408) ~[spring-boot-2.7.16.jar:2.7.16]
20.10.2023
9:57:51
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:307) ~[spring-boot-2.7.16.jar:2.7.16]
20.10.2023
9:57:51
	at org.springframework.boot.builder.SpringApplicationBuilder.run(SpringApplicationBuilder.java:164) ~[spring-boot-2.7.16.jar:2.7.16]
20.10.2023
9:57:51
	at org.airsonic.player.Application.main(Application.java:70) ~[classes/:na]
20.10.2023
9:57:51
Caused by: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'playerService': Invocation of init method failed; nested exception is org.springframework.dao.InvalidDataAccessApiUsageException: No EntityManager with actual transaction available for current thread - cannot reliably process 'remove' call; nested exception is javax.persistence.TransactionRequiredException: No EntityManager with actual transaction available for current thread - cannot reliably process 'remove' call
20.10.2023
9:57:51
	at org.springframework.beans.factory.annotation.InitDestroyAnnotationBeanPostProcessor.postProcessBeforeInitialization(InitDestroyAnnotationBeanPostProcessor.java:160) ~[spring-beans-5.3.30.jar:5.3.30]
20.10.2023
9:57:51
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.applyBeanPostProcessorsBeforeInitialization(AbstractAutowireCapableBeanFactory.java:440) ~[spring-beans-5.3.30.jar:5.3.30]
20.10.2023
9:57:51
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1796) ~[spring-beans-5.3.30.jar:5.3.30]
20.10.2023
9:57:51
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:620) ~[spring-beans-5.3.30.jar:5.3.30]
20.10.2023
9:57:51
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:542) ~[spring-beans-5.3.30.jar:5.3.30]
20.10.2023
9:57:51
	at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:335) ~[spring-beans-5.3.30.jar:5.3.30]
20.10.2023
9:57:51
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:234) ~[spring-beans-5.3.30.jar:5.3.30]
20.10.2023
9:57:51
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:333) ~[spring-beans-5.3.30.jar:5.3.30]
20.10.2023
9:57:51
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:208) ~[spring-beans-5.3.30.jar:5.3.30]
20.10.2023
9:57:51
	at org.springframework.beans.factory.config.DependencyDescriptor.resolveCandidate(DependencyDescriptor.java:276) ~[spring-beans-5.3.30.jar:5.3.30]
20.10.2023
9:57:51
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.doResolveDependency(DefaultListableBeanFactory.java:1391) ~[spring-beans-5.3.30.jar:5.3.30]
20.10.2023
9:57:51
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.resolveDependency(DefaultListableBeanFactory.java:1311) ~[spring-beans-5.3.30.jar:5.3.30]
20.10.2023
9:57:51
	at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor$AutowiredFieldElement.resolveFieldValue(AutowiredAnnotationBeanPostProcessor.java:710) ~[spring-beans-5.3.30.jar:5.3.30]
20.10.2023
9:57:51
	... 19 common frames omitted
20.10.2023
9:57:51
Caused by: org.springframework.dao.InvalidDataAccessApiUsageException: No EntityManager with actual transaction available for current thread - cannot reliably process 'remove' call; nested exception is javax.persistence.TransactionRequiredException: No EntityManager with actual transaction available for current thread - cannot reliably process 'remove' call
20.10.2023
9:57:51
	at org.springframework.orm.jpa.EntityManagerFactoryUtils.convertJpaAccessExceptionIfPossible(EntityManagerFactoryUtils.java:403) ~[spring-orm-5.3.30.jar:5.3.30]
20.10.2023
9:57:51
	at org.springframework.orm.jpa.vendor.HibernateJpaDialect.translateExceptionIfPossible(HibernateJpaDialect.java:235) ~[spring-orm-5.3.30.jar:5.3.30]
20.10.2023
9:57:51
	at org.springframework.orm.jpa.AbstractEntityManagerFactoryBean.translateExceptionIfPossible(AbstractEntityManagerFactoryBean.java:551) ~[spring-orm-5.3.30.jar:5.3.30]
20.10.2023
9:57:51
	at org.springframework.dao.support.ChainedPersistenceExceptionTranslator.translateExceptionIfPossible(ChainedPersistenceExceptionTranslator.java:61) ~[spring-tx-5.3.30.jar:5.3.30]
20.10.2023
9:57:51
	at org.springframework.dao.support.DataAccessUtils.translateIfNecessary(DataAccessUtils.java:243) ~[spring-tx-5.3.30.jar:5.3.30]
20.10.2023
9:57:51
	at org.springframework.dao.support.PersistenceExceptionTranslationInterceptor.invoke(PersistenceExceptionTranslationInterceptor.java:152) ~[spring-tx-5.3.30.jar:5.3.30]
20.10.2023
9:57:51
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186) ~[spring-aop-5.3.30.jar:5.3.30]
20.10.2023
9:57:51
	at org.springframework.data.jpa.repository.support.CrudMethodMetadataPostProcessor$CrudMethodMetadataPopulatingMethodInterceptor.invoke(CrudMethodMetadataPostProcessor.java:145) ~[spring-data-jpa-2.7.16.jar:2.7.16]
20.10.2023
9:57:51
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186) ~[spring-aop-5.3.30.jar:5.3.30]
20.10.2023
9:57:51
	at org.springframework.aop.interceptor.ExposeInvocationInterceptor.invoke(ExposeInvocationInterceptor.java:97) ~[spring-aop-5.3.30.jar:5.3.30]
20.10.2023
9:57:51
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186) ~[spring-aop-5.3.30.jar:5.3.30]
20.10.2023
9:57:51
	at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:241) ~[spring-aop-5.3.30.jar:5.3.30]
20.10.2023
9:57:51
	at jdk.proxy2/jdk.proxy2.$Proxy218.deleteAllByNameIsNullAndClientIdIsNullAndLastSeenBefore(Unknown Source) ~[na:na]
20.10.2023
9:57:51
	at org.airsonic.player.service.PlayerService.deleteOldPlayers(PlayerService.java:115) ~[classes/:na]
20.10.2023
9:57:51
	at org.airsonic.player.service.PlayerService.init_aroundBody0(PlayerService.java:88) ~[classes/:na]
20.10.2023
9:57:51
	at org.airsonic.player.service.PlayerService$AjcClosure1.run(PlayerService.java:1) ~[classes/:na]
20.10.2023
9:57:51
	at org.springframework.transaction.aspectj.AbstractTransactionAspect.ajc$around$org_springframework_transaction_aspectj_AbstractTransactionAspect$1$2a73e96cproceed(AbstractTransactionAspect.aj:67) ~[spring-aspects-5.3.30.jar:5.3.30]
20.10.2023
9:57:51
	at org.springframework.transaction.aspectj.AbstractTransactionAspect$AbstractTransactionAspect$1.proceedWithInvocation(AbstractTransactionAspect.aj:73) ~[spring-aspects-5.3.30.jar:5.3.30]
20.10.2023
9:57:51
	at org.springframework.transaction.interceptor.TransactionAspectSupport.invokeWithinTransaction(TransactionAspectSupport.java:388) ~[spring-tx-5.3.30.jar:5.3.30]
20.10.2023
9:57:51
	at org.springframework.transaction.aspectj.AbstractTransactionAspect.ajc$around$org_springframework_transaction_aspectj_AbstractTransactionAspect$1$2a73e96c(AbstractTransactionAspect.aj:71) ~[spring-aspects-5.3.30.jar:5.3.30]
20.10.2023
9:57:51
	at org.airsonic.player.service.PlayerService.init(PlayerService.java:86) ~[classes/:na]
20.10.2023
9:57:51
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:na]
20.10.2023
9:57:51
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[na:na]
20.10.2023
9:57:51
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[na:na]
20.10.2023
9:57:51
	at java.base/java.lang.reflect.Method.invoke(Unknown Source) ~[na:na]
20.10.2023
9:57:51
	at org.springframework.beans.factory.annotation.InitDestroyAnnotationBeanPostProcessor$LifecycleElement.invoke(InitDestroyAnnotationBeanPostProcessor.java:389) ~[spring-beans-5.3.30.jar:5.3.30]
20.10.2023
9:57:51
	at org.springframework.beans.factory.annotation.InitDestroyAnnotationBeanPostProcessor$LifecycleMetadata.invokeInitMethods(InitDestroyAnnotationBeanPostProcessor.java:333) ~[spring-beans-5.3.30.jar:5.3.30]
20.10.2023
9:57:51
	at org.springframework.beans.factory.annotation.InitDestroyAnnotationBeanPostProcessor.postProcessBeforeInitialization(InitDestroyAnnotationBeanPostProcessor.java:157) ~[spring-beans-5.3.30.jar:5.3.30]
20.10.2023
9:57:51
	... 31 common frames omitted
20.10.2023
9:57:51
Caused by: javax.persistence.TransactionRequiredException: No EntityManager with actual transaction available for current thread - cannot reliably process 'remove' call
20.10.2023
9:57:51
	at org.springframework.orm.jpa.SharedEntityManagerCreator$SharedEntityManagerInvocationHandler.invoke(SharedEntityManagerCreator.java:299) ~[spring-orm-5.3.30.jar:5.3.30]
20.10.2023
9:57:51
	at jdk.proxy2/jdk.proxy2.$Proxy198.remove(Unknown Source) ~[na:na]
20.10.2023
9:57:51
	at org.springframework.data.jpa.repository.query.JpaQueryExecution$DeleteExecution.doExecute(JpaQueryExecution.java:277) ~[spring-data-jpa-2.7.16.jar:2.7.16]
20.10.2023
9:57:51
	at org.springframework.data.jpa.repository.query.JpaQueryExecution.execute(JpaQueryExecution.java:90) ~[spring-data-jpa-2.7.16.jar:2.7.16]
20.10.2023
9:57:51
	at org.springframework.data.jpa.repository.query.AbstractJpaQuery.doExecute(AbstractJpaQuery.java:156) ~[spring-data-jpa-2.7.16.jar:2.7.16]
20.10.2023
9:57:51
	at org.springframework.data.jpa.repository.query.AbstractJpaQuery.execute(AbstractJpaQuery.java:144) ~[spring-data-jpa-2.7.16.jar:2.7.16]
20.10.2023
9:57:51
	at org.springframework.data.repository.core.support.RepositoryMethodInvoker.doInvoke(RepositoryMethodInvoker.java:137) ~[spring-data-commons-2.7.16.jar:2.7.16]
20.10.2023
9:57:51
	at org.springframework.data.repository.core.support.RepositoryMethodInvoker.invoke(RepositoryMethodInvoker.java:121) ~[spring-data-commons-2.7.16.jar:2.7.16]
20.10.2023
9:57:51
	at org.springframework.data.repository.core.support.QueryExecutorMethodInterceptor.doInvoke(QueryExecutorMethodInterceptor.java:160) ~[spring-data-commons-2.7.16.jar:2.7.16]
20.10.2023
9:57:51
	at org.springframework.data.repository.core.support.QueryExecutorMethodInterceptor.invoke(QueryExecutorMethodInterceptor.java:139) ~[spring-data-commons-2.7.16.jar:2.7.16]
20.10.2023
9:57:51
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186) ~[spring-aop-5.3.30.jar:5.3.30]
20.10.2023
9:57:51
	at org.springframework.data.projection.DefaultMethodInvokingMethodInterceptor.invoke(DefaultMethodInvokingMethodInterceptor.java:76) ~[spring-data-commons-2.7.16.jar:2.7.16]
20.10.2023
9:57:51
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186) ~[spring-aop-5.3.30.jar:5.3.30]
20.10.2023
9:57:51
	at org.springframework.transaction.interceptor.TransactionInterceptor$1.proceedWithInvocation(TransactionInterceptor.java:123) ~[spring-tx-5.3.30.jar:5.3.30]
20.10.2023
9:57:51
	at org.springframework.transaction.interceptor.TransactionAspectSupport.invokeWithinTransaction(TransactionAspectSupport.java:388) ~[spring-tx-5.3.30.jar:5.3.30]
20.10.2023
9:57:51
	at org.springframework.transaction.interceptor.TransactionInterceptor.invoke(TransactionInterceptor.java:119) ~[spring-tx-5.3.30.jar:5.3.30]
20.10.2023
9:57:51
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186) ~[spring-aop-5.3.30.jar:5.3.30]
20.10.2023
9:57:51
	at org.springframework.dao.support.PersistenceExceptionTranslationInterceptor.invoke(PersistenceExceptionTranslationInterceptor.java:137) ~[spring-tx-5.3.30.jar:5.3.30]
20.10.2023
9:57:51
	... 53 common frames omitted
20.10.2023
9:57:51
Container stopped
```

--- a/airsonic-main/src/main/java/org/airsonic/player/dao/MediaFileDao.java
+++ b/airsonic-main/src/main/java/org/airsonic/player/dao/MediaFileDao.java
@@ -112,7 +112,7 @@ public class MediaFileDao extends AbstractDao {
      */
     public List<MediaFile> getChildrenOf(String path, int folderId, boolean onlyPresent, boolean noIndexedTracks) {
         return query("select " + QUERY_COLUMNS + " from media_file where parent_path=? and folder_id=?" +
-                        (onlyPresent ? " and present" : "") + (noIndexedTracks ? " and start_position < 0" : ""), rowMapper, path, folderId);
+                        (onlyPresent ? " and present" : "") + (noIndexedTracks ? " and start_position < 0" : "") + " order by start_position", rowMapper, path, folderId);
     }
 
     public List<MediaFile> getFilesInPlaylist(int playlistId) {

--- a/airsonic-main/src/main/java/org/airsonic/player/service/PlayerService.java
+++ b/airsonic-main/src/main/java/org/airsonic/player/service/PlayerService.java
@@ -36,12 +36,13 @@ import org.apache.commons.lang3.StringUtils;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.context.event.ApplicationReadyEvent;
 import org.springframework.context.annotation.DependsOn;
+import org.springframework.context.event.EventListener;
 import org.springframework.stereotype.Service;
 import org.springframework.transaction.annotation.Transactional;
 import org.springframework.web.bind.ServletRequestUtils;
 
-import javax.annotation.PostConstruct;
 import javax.servlet.http.Cookie;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
@@ -82,9 +83,8 @@ public class PlayerService {
     @Autowired
     private PlayerDaoPlayQueueFactory playerDaoPlayQueueFactory;
 
-    @PostConstruct
-    public void init() {
-        // TODO: configurable
+    @EventListener
+    public void onApplicationEvent(ApplicationReadyEvent event) {
         deleteOldPlayers(60);
     }
 
@@ -113,6 +113,7 @@ public class PlayerService {
         playerRepository.deleteAllByNameIsNullAndClientIdIsNullAndLastSeenIsNull();
         Instant lastSeen = Instant.now().minus(days, ChronoUnit.DAYS);
         playerRepository.deleteAllByNameIsNullAndClientIdIsNullAndLastSeenBefore(lastSeen);
+        LOG.info("Complete Deleting old players");
     }
 
     public Player getPlayer(HttpServletRequest request, HttpServletResponse response, String username) throws Exception {

